bc/channel/utils/connectors/masto.py

```diff
@@ -29,22 +30,42 @@ def get_api_object(self, version=None) -> ApiWrapper:
 
         return mastodon
 
+    def upload_media(self, media, alt_text) -> int:
+        media_dict = self.api.media_post(
+            media,
+            mime_type="image/png",
+            focus=(0, 1),
+            description=shorten(
+                alt_text,
+                width=1500,
+                placeholder="…",
+            ),
+        )
+        return media_dict["id"]
+
     def add_status(
-        self, message: str, text_image: TextImage | None = None
+        self,
+        message: str,
+        text_image: TextImage | None = None,
+        thumbnails: list[IO[bytes]] | None = None,
     ) -> int:
         media_ids = None
         if text_image:
-            media_dict = self.api.media_post(
+            media_id = self.upload_media(
                 text_image.to_bytes(),
-                mime_type="image/png",
-                focus=(0, 1),
-                description=shorten(
-                    f"An image of the entry's full text: {text_image.description}",
-                    width=1500,
-                    placeholder="…",
-                ),
+                f"An image of the entry's full text: {text_image.description}",
             )
-            media_ids = [media_dict["id"]]
+            media_ids = [media_id]
+
+        if thumbnails:
+            for idx, thumbnail in enumerate(thumbnails):
+                media_id = self.upload_media(
+                    thumbnail, f"Thumbnail of page {idx + 1}"
+                )
+                if media_ids:
+                    media_ids.append(media_id)
+                else:
+                    media_ids = [media_id]
```

Comment:
Why not make medea_ids an empty list up above, and you can simplify this?

Reply:
I thought the API could throw an error if I used an empty list but I tried your suggestion and It works fine.